# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/abbottabad123/flyrank-ml-track/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


I chose the Refresh / Content Opportunity Scoring lane because I want to identify content pages that may deserve review or refresh based on observable search and content-performance signals. This lane is useful because editors and SEO teams need to prioritize a large number of pages, and a ranked opportunity score can help them decide which pages to review first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The decision is which content pages should be reviewed or refreshed first. Editors and SEO teams would use the ranked recommendations to prioritize their work. A false positive could lead to unnecessary editorial effort, while a false negative could cause a potentially valuable refresh opportunity to be missed. The analysis will therefore be used as decision support rather than as an automatic decision-maker.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [10]:
import pandas as pd

path = "/content/flyrank-ml-track/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(path)

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Declining pages:", (df["trend_direction"] == "down").sum())

Rows: 30000
Columns: 44
Declining pages: 16262


The starter dataset contains 30,000 content-page records and 44 columns. Among these records, 16,262 pages are labeled as declining in the current performance window. This suggests that there is enough variation in the starter data to investigate which observable signals are associated with content refresh opportunities.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This work will provide observed and measured evidence about patterns in search and content-performance data. The final recommendations will be directional and intended for decision support, helping editors prioritize pages for review or refresh. The analysis will not prove causal effects, will not establish how Google's ranking algorithm works, and will not claim to predict Google's algorithm. Any model results will be interpreted as evidence from the available data rather than as causal proof.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.